rust-runtime/aws-smithy-http-server-python/src/error.rs

```diff
@@ -63,7 +63,12 @@ impl PyMiddlewareException {
 
 impl From<PyErr> for PyMiddlewareException {
     fn from(other: PyErr) -> Self {
-        Self::newpy(other.to_string(), None)
+        // Try to extract `PyMiddlewareException` from `PyErr` and use that if succeed
+        let middleware_err = Python::with_gil(|py| other.to_object(py).extract::<Self>(py));
+        match middleware_err {
+            Ok(err) => err,
+            Err(_) => Self::newpy(other.to_string(), None),
```

Comment:
Here we probably want a log statement capturing the details of the exception that we failed to extract since it didn't match the one we expected.

Reply:
The error on the `Err(_)` variant will always be `'TypeError' object cannot be converted to 'MiddlewareException'` here because it is a conversion error and the real error will be logged at https://github.com/awslabs/smithy-rs/blob/e12136563d734e3c1ea40d96a1705d3b1f5d9cbe/rust-runtime/aws-smithy-http-server-python/src/middleware/layer.rs#L124